elastic/apm-server
# aggregation/txmetrics: aggregate user_agent.name

## Motivation/summary

For RUM page-load transactions, parse the User-Agent
header and aggregate on the user_agent.name value.

We store the results of parsing User-Agent in an LRU
cache to minimise overhead. The size of the cache
defaults to 5000 entries, and can be controlled by
setting `apm-server.aggregation.rum.user_agent.lru_size`.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~

## How to test these changes

1. Enable aggregation (`apm-server.aggregation.enabled=true`)
2. Send RUM transactions, with multiple browsers (e.g. Chrome and Firefox)
3. Observe metrics produced with `user_agent.name` set to the relevant browser name

## Related issues

Closes #3814
